src/SdkCommon/ClientRuntime/ClientRuntime/ExtensibleEnum.cs

```diff
@@ -5,73 +5,77 @@
         {
-            _name = name;
+            _value = value;
         }
 
         /// <summary>
         /// Compares two ExtensibleEnum values for equality.
         /// </summary>
-        /// <param name="lhs">The first ExtensibleEnum to compare.</param>
-        /// <param name="rhs">The second ExtensibleEnum to compare.</param>
+        /// <param value="lhs">The first ExtensibleEnum to compare.</param>
+        /// <param value="rhs">The second ExtensibleEnum to compare.</param>
         /// <returns>true if the ExtensibleEnum objects are equal or are both null; false otherwise.</returns>
-        public static bool operator ==(ExtensibleEnum<T> lhs, T rhs) => object.Equals(lhs, rhs);
+        public static bool operator ==(ExtensibleEnum<T, V> lhs, T rhs) => object.Equals(lhs, rhs);
 
         /// <summary>
         /// Compares two ExtensibleEnum values for inequality.
         /// </summary>
-        /// <param name="lhs">The first ExtensibleEnum to compare.</param>
-        /// <param name="rhs">The second ExtensibleEnum to compare.</param>
+        /// <param value="lhs">The first ExtensibleEnum to compare.</param>
+        /// <param value="rhs">The second ExtensibleEnum to compare.</param>
         /// <returns>true if the ExtensibleEnum objects are not equal; false otherwise.</returns>
-        public static bool operator !=(ExtensibleEnum<T> lhs, T rhs) => !object.Equals(lhs, rhs);
+        public static bool operator !=(ExtensibleEnum<T, V> lhs, T rhs) => !object.Equals(lhs, rhs);
 
         /// <summary>
         /// Defines explicit conversion from ExtensibleEnum to string.
         /// </summary>
-        /// <param name="name">ExtensibleEnum to convert.</param>
+        /// <param val="val">ExtensibleEnum to convert.</param>
         /// <returns>The ExtensibleEnum as a string.</returns>
-        public static explicit operator string(ExtensibleEnum<T> name) => name?.ToString();
+        public static explicit operator string(ExtensibleEnum<T, V> val) => val?.ToString();
 
         /// <summary>
-        /// Looks up an ExtensibleEnum instance by name, or returns null if the given name does not match one of the
+        /// Looks up an ExtensibleEnum instance by value, or returns null if the given value does not match one of the
         /// known values of this ExtensibleEnum.
         /// </summary>
-        /// <param name="name">Name of the ExtensibleEnum value.</param>
-        /// <returns>An instance of type T with the given name, or null if no such instance could be found.</returns>
-        protected static T Lookup(string name)
+        /// <param value="value">value of the ExtensibleEnum value.</param>
+        /// <returns>An instance of type T with the given value, or null if no such instance could be found.</returns>
+
+        protected static T Lookup(V value)
+        //protected static T this[V value] 
         {
-            if (name == null)
+            if (value == null)
```

Comment:
There is no constraint that `V` is a reference type. How does this compile?

Reply:
Hmm... I don't think I saw any compilation issues, but I guess we can check for a nullable type before comparison.